assertion/function/assertiontree/backprop.go

```diff
@@ -602,10 +613,26 @@ buildShadowMask:
 							rootNode.addProductionsForAssignmentFields(fieldProducers, lhsVal)
 						}
 
+						// beforeLastIndex and afterLastIndex are used to find the newly added triggers
+						beforeLastIndex := len(rootNode.triggers)
+
 						rootNode.AddProduction(&annotation.ProduceTrigger{
 							Annotation: rproducers[0].GetShallow().Annotation,
 							Expr:       lhsVal,
 						}, rproducers[0].GetDeepSlice()...)
+
+						afterLastIndex := len(rootNode.triggers)
+
+						// Update consumers of newly added triggers with assignment entries for informative printing of errors
+						if len(rootNode.triggers) > beforeLastIndex && len(rootNode.triggers) <= afterLastIndex {
```

Comment:
Do we need this check?

(1) will `len(rootNode.triggers) > beforeLastIndex` ever be `false`? i.e., will `rootNode.triggers` actually shrink?

(2) `len(rootNode.triggers) == afterLastIndex` here, right?

Reply:
You are right, revised the checks.